python-project-template/.github/workflows/{% if mypy_type_checking != 'none' %}type-checking.yml{% endif %}.jinja

```diff
@@ -0,0 +1,35 @@
+# This workflow will install Python dependencies, then perform static type checking analysis.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: mypy Type checking
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
```

Comment:
Not necessary to make a change here, but would it be useful to cover 3.8-3.10? Some of the other workflows will cover those python versions, but at least one existing workflow only covers 3.10.